### DISCUSSION TITLE
Tasks/Preparation for GDCR24

### DISCUSSION THREAD

**Discussion by codecop** [swkWien/sessions#140] [General]
@rolger @maosmurf @gregorriegler and me:
Based on our discussion 20241011 and the decisions there made, we need to prepare the following things (based on the time table I created)

* [x] Intro	09:00	09:10	Welcome, organisation, fire exit, etc. > @rolger 
* [x] Game	09:10	09:20	Connect/Standing Survey: Mindset > @rolger
* [x] Intro	09:20	09:30	Intro topic, day layout and timetable > Peter
* [x] Session 09:30	10:30	Ensemble: TDD/kata Intro, by language	> all facilitators including @caichinger who will support us before noon. Maybe we can find more ppl
  * [x] find suitable rooms for maybe 6 groups > @rolger 
  * [x] Because also some people from participants will have to run Ensemble we need a handout for the Ensemble facilitator (with basic rules etc. See notes for session in #136  > TODO 
  * [x] Retro	10:30	10:40	Session Retro	> all - TODO What do we want to do here?
* [x] Intro	10:50	10:55	Define Incremental > @gregorriegler
* [x] Session	10:55	11:50	second Ensemble: Incremental Intro, new groups	> all facilitators again.
  * [x] second handout for Ensemble facilitator with help/ideas about incremental dev > TODO
  * [x] Retro	11:50	12:00	Session Retro	> all - TODO What do we want to do here?
* [x] Game	13:00	13:10	Connect/Activation > @rolger 
* [x] Session	13:10	14:10	Constraint: Baby Steps, finding pairs, Session, Retro > TODO (single facilitator) @rolger 
* [x] Session 14:40		15:45 	Constraint: TCR, including *Demo* (important) Session, Retro > TODO (single facilitator) @gregorriegler 
* [ ] Session 	15:55	17:00	Constraint: Evil Ping Pong, including *Demo* (important) Session, Retro > TODO (single facilitator) @maosmurf 
* [x] Closing Retro	17:00	17:30	Closing Retro in smaller groups	> @rolgerm

**Comment by codecop** [comment]
Habe mal die Eckpunkte die wir entschieden haben zusammen gefasst

Motto für uns (Facilitators): Teaching Iterative Development

# Sessions

## 1. Ensemble: TDD kata Intro, by languages
Gruppen nach gemeinsamen Sprachen bilden

Ziel
* Ensemble Modus kennen lernen
* TDD kennen lernen
* Problem/Kata kennen lernen

Facilitation
* Ensemble strikt geführt mit Rotation
* Inkremente nach Wunsch der Teilnehmer
* (vermutlich zu groß, führt zu Lerneffekt bei 2. Session)

Handout für Ensemble Facilitators
* Ensemble Rules, Rotation auf Zeit
* Ensemble Facilitation Tipps
* 3 Laws of TDD
* Link Wikipedia Tetris

## 2a. Incremental erklären

* Bild inkrementell vs. iterativ
* "Es läuft jederzeit etwas Sinnvolles"
* Was ist ein guter erster Testfall
* Wie man ein größeres Element zerteilt
* zB eine Dimension

## 2b. Ensemble: Incremental Dev Intro
neue Gruppen mit anderen Facilitators

Ziel
* wollen möglichst kleine Schritte machen

Handout für Facilitators
* Was ist der minimale Business Value
* Was kann ich noch weglassen
* Ablauf/Beispiel Tests nach Session 12.10.

## Lunch

## 3. Baby Steps (TDD, Pair Programming, Timer)
* Handout für Paare (DONE)

## 4. TCR without tools (Pair Programming)
* Demo (auch Tests verschwinden bei Revert)
* Handout für Paare (DONE)

**Comment by codecop** [comment]
I have pushed my material so far here https://github.com/swkWien/sessions/tree/main/2024/cr20241109_gdrc

**Comment by gregorriegler** [comment]
# GDCR 2024 Vorbereitung, 2024-10-25 Notes

# Ensemble Moderatoren:
5 Personen je Gruppe (30 Gäste wäre 6 Facilitators)
- Peter
- Gregor (Java od C#)
- Adam (Gerne)
- Claus Aichinger
- Georg Ebner ?
- Andreas Kleinbichler
[ ] Roland leitet nächsdten Termin an Andreas weiter
- Jakob Durstbeger
- Roland (Eher nicht, weil host)

# Peters Handouts für Ensemble Moderatoren
[ ] Anschauen, füllen: https://github.com/swkWien/sessions/tree/main/2024/cr20241109_gdrc

# 6 Ensembles
- Wir haben alle Meeting Räume mit Laptop
- Idee: Welcher Raum hat welche Sprache
- Ensembles rotieren ihren Moderator

## Je Ensemble
- Tastatur + Maus, Gerät mit TDD Setup in einer Sprache
	- sodass alles schnell funktioniert

# Abklären definition von Incremental
## Blog
https://blog.crisp.se/2016/01/25/henrikkniberg/making-sense-of-mvp
## Bild
https://blog.crisp.se/wp-content/uploads/2016/01/mvp.png
[ ] Gregor bereitet Content und Slide vor

# Classic Nachmittag
## Baby Steps, Nach 2 minuten nicht grün = revert.
https://github.com/swkWien/sessions/blob/main/2024/cr20241109_gdrc/Constraint%20-%20Baby%20Steps%20(Print).pdf 
Wenn es gut funktioniert mit timer runter gehen (1 min, 30 sec)

# Ausdrucken 
[ ] Adam Druckt gerne.

# Slides
[ ] Roland macht ne Kopie Slides vom letzten Jahr mit Slides 
		- nach Zeitplan

# Zeitplan
![https://private-user-images.githubusercontent.com/830028/376205238-8be81a87-84bf-4e5c-9be4-7f628415de5f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4NjYzMTMsIm5iZiI6MTcyOTg2NjAxMywicGF0aCI6Ii84MzAwMjgvMzc2MjA1MjM4LThiZTgxYTg3LTg0YmYtNGU1Yy05YmU0LTdmNjI4NDE1ZGU1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNVQxNDIwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZDM0OGRkYjFlNzc0ZDljOGFjMDE3ZjRhNjYwMjA2NDM2NjMyZGYwZjhlZDAxMTVlY2E2YTNmODEyNTllMTQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6Q-Hi9Q3PPju6aXGbQbMHrNIyY9Jf7SnEHNgzdrXRl8]()

## Babysteps
[ ] Roland bereitet Babysteps vor (oder einer der anderen neuen)

## TCR wird zweite Session (Gregor würds gerne machen)
[ ] Gregor bereitet TCR Session vor

## Evil Ping Pong am Schluss (Adam würds gerne facilitieren), 
		alternativ TDDaymi am schluss
[ ] Adam bereitet Evil Ping Pong vor

#
[ ] Wo geben wir die learnings hin?

## Nächster Termin
[ ] Ensemble Facilitation guideline gemeinsam machen
[ ] Slides nochmal durchgehen

**Comment by rolger** [comment]
Ihr findet das Grundgerüst für die Slides in diesem Repo https://github.com/swkWien/gdcr2024. @gregorriegler hat eine erste Version mit reveal.js erstellt. Mit reveal.js ist es sehr einfach lokal zu testen s. README.

Ich habe einen PR mit einer Unterteilung nach Sessions erstellt. Außerdem habe ich das Deployment getestet:
[swkwien.github.io/gdcr2024/](https://swkwien.github.io/gdcr2024/)

**Reply by gregorriegler** [reply]
Super Roland 👍. Wenn ich könnte würde ich deinen PR mergen. Ich denk es wäre am sinnvollsten wenn du uns einlädst und wir einfach auf main direkt pushen.

**Reply by codecop** [reply]
Danke Slide Setup schaut gut aus, lokaler Modus funkioniert.

**Reply by rolger** [reply]
ich habe euch alle eingeladen, ihr solltet alle direkt ändern können

**Reply by gregorriegler** [reply]
Habe keinen invite

**Reply by rolger** [reply]
es sollte einfach funktionieren. ich das ganze Team [Organisers](https://github.com/orgs/swkWien/teams/organisers) und [GDCR24](https://github.com/orgs/swkWien/teams/gdcr24) hinzugefügt

**Comment by rolger** [comment]
ich hatte eine Diskussion über inkrementell und iterativ und meine Bedenken zu unseren Slides. ich würde gerne darüber reden, sollen wir das am Montag machen?

**Reply by gregorriegler** [reply]
Gerne, habe noch ideen was ich ändern und ergänzen will. Ich will die slide mit dem meme entfernen und mehr auf feedback und die Psychologie von feedback-losigkeit eingehen (e.g. Overshooting)

**Reply by gregorriegler** [reply]
Kannst du uns bitte zum repo Berechtigung geben

**Reply by rolger** [reply]
hast du nicht? es sollte funktionieren

**Reply by caichinger** [reply]
Refs #131

**Comment by codecop** [comment]
@rolger Deine Bilder fehlen in den Slides.
Ich habe die Slides etwas überarbeitet und eingefügt wo ich etwas gewusst habe. Connect/Introduce Spiel ist fertig. Intro sieht fertig aus. Session 3 (Baby Steps) sieht fertig aus.

**Comment by gregorriegler** [comment]
Ich wollte mit gitpod.io arbeiten, kann aber von dort nicht pushen.
Ärgerlich, da ich eigentlich persönlich die Berechtigung habe auf das Repository zu pushen.
Anscheinend ist dies durch die Organisatzion swkWien blockiert und muss erst freigegeben werden.
Das kann ich aber nicht machen, da ich keine Berechtigung auf der Organisation habe.
Das kostet mich gerade Zeit und Nerven.

**Reply by codecop** [reply]
Hab's approved

**Comment by caichinger** [comment]
### Bitte Check meines Rollenverständnisses

Ich fasse hier mein Verständnis meiner Rolle und der dafür notwendigen Infos zusammen, falls da was nicht stimmt, bitte um Hilfe:

* Ich unterstütze bei Facilitation von Session 1 + 2, beide Sessions sind als Ensemble gedacht
* Kata: Tetris
* Session 1: TDD/Kata Intro
* Session 2: Incremental Dev Intro
* Facilitation Advice kommt noch von Peter

@codecop wenn du die Infos hast und daran denkst, schreib mir bitte auf Signal. Geht sonst in den Notifications unter. Danke.

---

Danke für eure Arbeit, glaube da ist wieder was spannendes im Werden!

**Reply by codecop** [reply]
Je genau danke Dir Claus. Es wird ein Handout geben dass an alle Facilitators verteilt wird. Brauchst nichts vorbereiten.

**Comment by codecop** [comment]
Görge ist auch als Ensemble Moderator dabei. D.h. Peter, Gregor, Adam, Claus, Görge. Fehlt uns noch mindestens einer. Jakob?

**Reply by rolger** [reply]
Ich frage morgen noch einmal nach

**Reply by codecop** [reply]
David Tanzer kommt und kann ein Ensemble übernehmen!

**Comment by rolger** [comment]
Daniel übernimmt die Moderation von der Retrospektive und dem Ice Breaker mittags. Er könnte auch die Runde am Anfang übernehmen.
Peter, übrigens heißt das Format Mad Tea Party, mir ist es gestern nicht eingefallen (https://liberatingstructures.de/mad-tea-party/)

**Comment by rolger** [comment]
Bitte um Reviews:
- Bilder für Session 3 sind alle aktualisiert
- Slides für Closing Circle sind fertig
- Slides für Activation sind fertig
- Pause überall eingefügt, weil include nicht funktioniert hat und Zeit auf 10 Minuten gekürzt
- Intro überarbeitet z.B. Timetable
- Let's Connect überarbeitet

**Reply by rolger** [reply]
@codecop mir ist aufgefallen, dass unser Connect in den Slides anders ist als im Zeitplan. 

Außerdem habe ich einen Vorschlag um den Connect anderes zu machen. Ich brauche 7-8 Minuten (1min - 2min - 4min). Ich würde wie am Ende 1-2-4-All ohne All machen, und dazu foldende Fragen verwenden: 
1. Wer bin ich ...
2. Was will ich mitnehmen ...
3. Was ist mir wichtig ...

Ich würde alle Teilnehmer in 2 Dimensionen aufstellen - Erfahrung / Programmiersprachen. Dann in 4 Quatranten aufteilen und sie wechselseitig zusammen kommen lassen. Somit kommt es zu einer bestmöglichen Mischung aus Erfahrung und Sprachen :-)

**Reply by codecop** [reply]
Ist mir auch Recht wenn du es machst. Wenn du beides machst braucht der Zeitplan keine Trennung von Intro und Connect. Gesamtzeit sollte halt gleich bleiben

**Comment by codecop** [comment]
Ich hab David erlaubt eine Freikarte für Socrates 25 zu verlosen. Wird vor closing sein. Er wird pull request mit Slide dazu schicken

**Comment by codecop** [comment]
Habe alle Slides nochmal kontrolliert und sage mal fertig.

**Comment by caichinger** [comment]
Danke an alle! War eine Freude dabei zu sein und habe gerade wieder große Lust auf CR!

Habe hier ein paar Retro-Notizen abgelegt: https://github.com/swkWien/sessions/discussions/143
